coral/src/domain/acl/acl-types.ts

```diff
@@ -0,0 +1,42 @@
+import { KlawApiRequest } from "types/utils";
+
+// Several types are dependent on topictype when it is "Consumer":
+// - aclPatternType can only be "LITERAL"
+// - consumergroup is required
+// - transactionalId becomes available
+// So we define two types depending on the topictype
+
+type CreateAclRequestTopicTypeProducer = Pick<
```

Comment:
As the two types share most of the properties, would be export those into its own type, which is not exported. I think that would make the code a bit DRYer and it is easier to spot the shared values

```
type CommonAclRequest = Pick<
  KlawApiRequest<"createAclRequest">,
  | "remarks"
  | "aclPatternType"
  | "topicname"
  | "environment"
  | "teamname"
  | "aclIpPrincipleType"
  | "consumergroup"
  | "acl_ssl"
  | "acl_ip"
>;
```

Reply:
Good point! I'm thinking of creating a `BaseAclRequest` that has all the possible properties of the request, and then create interesection types which narrow it down for the consumer and producer requests?

```ts
type BaseAclRequest = Pick<
  KlawApiRequest<"createAclRequest">,
  | "remarks"
  | "aclPatternType"
  | "topictype"
  | "topicname"
  | "environment"
  | "teamname"
  | "aclIpPrincipleType"
  | "consumergroup"
  | "acl_ssl"
  | "acl_ip"
>;

type CreateAclRequestTopicTypeProducer = BaseAclRequest & {
  topictype: "Producer";
};

type CreateAclRequestTopicTypeConsumer = BaseAclRequest & {
  topictype: "Consumer";
  aclPatternType: "LITERAL";
  consumergroup: string;
};

```